### ISSUE TITLE
andOperator rule breaks where condition on repeat block

### ISSUE THREAD

**Issue by ileitch** [nicklockwood/SwiftFormat#416]
Given the following code:

```swift
repeat {
    print("hello")
} while true && !false
```

The `andOperator` rule will produce invalid syntax:

```swift
repeat {
    print("hello")
} while true, !false
```

**Comment by nicklockwood** [comment]
@ileitch thanks for reporting this, but I can't reproduce the issue with the sample you've given me. Are you using the latest version of SwiftFormat? 

If so, maybe the bug only happens when there is some additional code before or after the loop? Can you provide a larger sample?

**Comment by nicklockwood** [comment]
@ileitch never mind, I've managed to reproduce it. It happens if there's a "{" at some point after the repeat/while statement.

**Comment by nicklockwood** [comment]
@ileitch fixed in 0.40.7
